connect/src/com/telenor/connect/ui/ConnectWebViewClient.java

```diff
@@ -94,6 +109,75 @@ public boolean shouldOverrideUrlLoading(WebView view, String url) {
     }
 
     @Override
+    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
+    public WebResourceResponse shouldInterceptRequest(WebView view, WebResourceRequest request) {
+        boolean shouldSkipInterceptionLogic
+                = ConnectSdk.getCellularNetwork() == null || ConnectSdk.getWifiNetwork() == null;
+        if (shouldSkipInterceptionLogic) {
+            return null;
+        }
+        if (shouldFetchThroughCellular(request.getUrl().toString())) {
+            return fetchUrlTroughCellular(request.getUrl().toString());
+        }
+        return null;
+    }
+
+    public boolean shouldFetchThroughCellular(String url) {
+        if (ConnectSdk.getWellKnownConfig() == null
+                || ConnectSdk.getWellKnownConfig().getNetworkAuthenticationTargetIps() == null
+                || ConnectSdk.getWellKnownConfig().getNetworkAuthenticationTargetIps().isEmpty()) {
+            return false;
+        }
+        String hostIp;
+        try {
+            String host = (new URL(url)).getHost();
+            hostIp = InetAddress.getByName(host).getHostAddress();
+        } catch (MalformedURLException|UnknownHostException e) {
+            return false;
+        }
+        return ConnectSdk
+                .getWellKnownConfig()
+                .getNetworkAuthenticationTargetIps()
```

Comment:
Just a control question... I do not see that you tackle the situation when wellKnowConfig() returns null, let alone getNetworkAuthenticationTargetIps() returning null. For operators other than us, it can happen, can't it?